plugins/modules/ndo_physical_interface.py

```diff
@@ -0,0 +1,404 @@
+                proposed_payload["breakoutMode"] = breakout_mode
+
+            if interfaces and interfaces != match.details.get("interfaces"):
+                ops.append(dict(op="replace", path=physical_policy_path + "/interfaces", value=interfaces))
+                proposed_payload["interfaces"] = interfaces
+
+            # Node changes are not reflected on UI
+            if interface_descriptions and match.details.get("interfaceDescriptions") != interface_descriptions:
+                updated_interface_descriptions = format_interface_descriptions(interface_descriptions, "")
+                ops.append(dict(op="replace", path=physical_policy_path + "/interfaceDescriptions", value=updated_interface_descriptions))
+                proposed_payload["interfaceDescriptions"] = updated_interface_descriptions
+            elif interface_descriptions == [] and match.details.get("interfaceDescriptions"):
+                ops.append(dict(op="remove", path=physical_policy_path + "/interfaceDescriptions"))
+
+            mso.sanitize(proposed_payload, collate=True)
+
+        else:
+            if not nodes:
+                mso.fail_json(msg=("ERROR: Missing 'nodes' for creating a Physical Interface."))
+
+            if not physical_interface_type:
+                mso.fail_json(msg=("ERROR: Missing Physical Interface type for creating a Physical Interface."))
+
+            payload = {
+                "name": name,
+                "templateId": mso_template.template.get("templateId"),
+                "schemaId": mso_template.template.get("schemaId"),
+                "nodes": nodes,
+                "interfaces": interfaces,
+                "policyGroupType": physical_interface_type,
+            }
+
+            if description:
+                payload["description"] = description
+
+            if physical_interface_type == "physical" and physical_policy_uuid:
+                payload["policy"] = physical_policy_uuid
+
+            if physical_interface_type == "breakout" and breakout_mode:
+                payload["breakoutMode"] = breakout_mode
+
+            if interface_descriptions:
+                payload["interfaceDescriptions"] = format_interface_descriptions(interface_descriptions, "")
+
+            mso.sanitize(payload)
+            ops.append(dict(op="add", path="/fabricResourceTemplate/template/interfaceProfiles/-", value=mso.sent))
+
+    elif state == "absent":
+        if match:
+            ops.append(dict(op="remove", path=physical_policy_path))
+
+    if not module.check_mode and ops:
+        response = mso.request(mso_template.template_path, method="PATCH", data=ops)
+        physical_interfaces = response.get("fabricResourceTemplate", {}).get("template", {}).get("interfaceProfiles", [])
+        mso.stdout = str(physical_interfaces)
+        match = mso_template.get_object_by_key_value_pairs(object_description, physical_interfaces, [KVPair("uuid", uuid) if uuid else KVPair("name", name)])
```

Comment:
Changed this function in template.py to return null when physical_interfaces is inputted as null. 

    def get_object_by_key_value_pairs(self, object_description, search_list, kv_list, fail_module=False):
        """
        Get the object from a list of mso object dictionaries by name.
        :param object_description: Description of the object to search for -> Str
        :param search_list: Objects to search through -> List.
        :param kv_list: Key/value pairs that should match in the object. -> List[KVPair(Str, Str)]
        :param fail_module: When match is not found fail the ansible module. -> Bool
        :return: The object. -> Dict | None
        """
        if search_list is None or kv_list is None:
            return None
        match, existing = self.get_object_from_list(search_list, kv_list)
        if not match and fail_module:
            msg = "Provided {0} with '{1}' not matching existing object(s): {2}".format(object_description, kv_list, ", ".join(existing))
            self.mso.fail_json(msg=msg)
        return match

Reply:
changed